src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -221,7 +221,7 @@ private bool TryRewriteTagHelperStart(MarkupStartTagSyntax startTag, out MarkupT
                 var tagName = startTag.GetTagName();
 
                 // Could not determine tag name, it can't be a TagHelper, continue on and track the element.
-                if (tagName == null)
+                if (tagName == null || tagName.StartsWith("!"))
```

Comment:
We were returning null tag name for bang tags before and so tag helpers didn't apply. Now that I am returning the correct name, I need to add this so tag helpers won't apply for bang tags.

Reply:
This seems right. You could also consider putting this on the start tag class.